packages/ui-library/src/components/sidebar-navigation/link.js

```diff
@@ -0,0 +1,48 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React, { useCallback } from "react";
+import { useNavigationContext } from "./index";
+
+/**
+ * @param {JSX.ElementClass} [as="a"] The component.
+ * @param {string} [pathProp="href"] The key of the path in the props.
+ * @param {?JSX.node} [children=null] The content.
+ * @param {string} [className] The classname.
+ * @param {Object} [props] Extra props.
+ * @returns {JSX.Element} The element.
+ */
+export const Link = ( { as: Component = "a", pathProp = "href", children = null, className, ...props } ) => {
+	const { activePath, setMobileMenuOpen } = useNavigationContext();
+
+	const handleClick = useCallback( () => {
+		setMobileMenuOpen( false );
+		// eslint-disable-next-line react/prop-types
+		if ( typeof props?.onClick === "function" ) {
```

Comment:
if the prop type would be a function then this check can be replaced for the check it's not null.

```suggestion
		if ( onClick  ) {
```

Reply:
Except that React is moving away from PropTypes. So I suspect this is better in the long run.
https://react.dev/blog/2024/04/25/react-19-upgrade-guide#removed-proptypes-and-defaultprops